source/hp/dof_level.cc

```diff
@@ -196,8 +197,14 @@ namespace internal
                 // apparently so. uncompress
                 Assert (next_offset-dof_offsets[cell] == 1,
                         ExcInternalError());
-                for (unsigned int i=0; i<fe_collection[active_fe_indices[cell]].template n_dofs_per_object<dim>(); ++i)
+                const unsigned int dofs_per_object
+                  = fe_collection[(active_fe_index_type)~(signed_active_fe_index_type)active_fe_indices[cell]]
```

Comment:
Can you create a function for this cast/bit flips that you are doing and give it a name that explains what it does.